main/streams/plain_wrapper.c

```diff
@@ -285,6 +285,8 @@ PHPAPI php_stream *_php_stream_fopen_from_fd(int fd, const char *mode, const cha
 		if (!self->is_seekable) {
 			stream->flags |= PHP_STREAM_FLAG_NO_SEEK;
 			stream->position = -1;
+		} else if (zero_position) {
+			stream->position = 0;
```

Comment:
What about adding an assertion here:
```suggestion
			ZEND_ASSERT(zend_lseek(self->fd, 0, SEEK_CUR) == 0);
			stream->position = 0;
```

Reply:
Amended the commit with this assertion.